src/components/HCPSignupFinish.js

```diff
@@ -1,114 +1,166 @@
+
 import Firebase from 'firebase';
 import * as firebaseui from 'firebaseui';
-import { withRouter } from 'react-router-dom';
+
+import FormBuilder from 'components/Form/FormBuilder';
+import { formFieldTypes } from 'components/Form/CreateFormFields';
 
 import { Routes } from 'constants/Routes';
 import { routeWithParams } from 'lib/utils/routes';
+import { isValidEmail } from 'lib/utils/validations';
 
-class HCPSignupFinish extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      email: '',
-      confirmEmail: false,
-      dropsite: this.props.match.params.id,
-    };
-    this.ui = new firebaseui.auth.AuthUI(Firebase.auth());
-
-    this.submitEmail = this.submitEmail.bind(this);
-    this.handleEmailChange = this.handleEmailChange.bind(this);
-  }
-
-  componentDidMount() {
-    const shouldConfirm = this.props.backend.shouldRepromptEmail();
-    this.setState({ confirmEmail: shouldConfirm });
-    if (shouldConfirm === false) {
-      this.submitEmail();
-    }
-  }
+import { useLoggedIn } from 'hooks/useLoggedIn';
+
+import Page from 'components/layouts/Page';
+import PageLoader from 'components/Loader/PageLoader';
+import Note from 'components/Note';
+import Anchor, { anchorTypes } from 'components/Anchor';
+
+function HCPSignupFinish({ backend }) {
+  const { loggedIn } = useLoggedIn();
+  const history = useHistory();
+  const params = useParams();
+  const { t } = useTranslation();
 
-  submitEmail(event) {
-    if (event) {
-      event.preventDefault();
+  const [email, setEmail] = useState('');
+  const [shouldConfirmEmail, setShouldConfirmEmail] = useState(false);
+  const [isLoading, setIsLoading] = useState(true);
+
+  const { id: dropsite } = params;
+
+  useEffect(() => {
+    if (loggedIn && dropsite) {
+      history.push(
+        routeWithParams(Routes.DROPSITE_ADMIN, {
+          id: dropsite,
+        }),
+      );
     }
-    let url = window.location.href;
-    let hasAddress;
-    this.props.backend
-      .continueSignup(url, this.state.confirmEmail ? this.state.email : null)
-      .then(() => {
-        this.props.backend
-          .getDropSites(this.state.dropsite)
-          .then((data) => {
-            hasAddress = !!data.dropSiteAddress;
-            if (!hasAddress) {
-              this.props.history.push(
-                routeWithParams(Routes.DROPSITE_NEW_ADMIN, {
-                  id: this.state.dropsite,
-                }),
-              );
-            }
-          })
-          .then(() => {
-            if (!hasAddress) {
-              return;
-            }
-            this.props.backend
-              .getRequests(this.state.dropsite)
-              .then((requests) => {
+  }, [loggedIn, history, dropsite]);
+
+  // todo: do we need this firebaseui.auth.AuthUI ?
+  // eslint-disable-next-line no-unused-vars
+  const ui =
+    firebaseui.auth.AuthUI.getInstance() ||
+    new firebaseui.auth.AuthUI(Firebase.auth());
```

Comment:
Pretty sure this isn't doing anything. Let's cut it?

Reply:
yeah this is probably copypasta from when I copied this from another login page